userspace/libsinsp/examples/test.cpp

```diff
@@ -38,254 +48,347 @@ static sinsp_evt_formatter* default_formatter = nullptr;
+				sinsp_fdinfo_t* fd_info = ev->get_fd_info();
+
+				// event subcategory should contain SC_NET if ipv4/ipv6
+				if(nullptr != fd_info && (fd_info->get_l4proto() != SCAP_L4_UNKNOWN && fd_info->get_l4proto() != SCAP_L4_NA))
+				{
+					cout << "[" << fd_info->tostring() << "]:";
+				}
+			}
+			else if(ev->get_category() == EC_IO_READ || ev->get_category() == EC_IO_WRITE)
+			{
+				cout << get_event_category(ev->get_category()) << "]:";
+
+				sinsp_fdinfo_t* fd_info = ev->get_fd_info();
+				if(nullptr != fd_info && (fd_info->get_l4proto() != SCAP_L4_UNKNOWN && fd_info->get_l4proto() != SCAP_L4_NA))
+				{
+					cout << "[" << fd_info->tostring() << "]:";
+				}
+			}
+			else
+			{
+				cout << get_event_category(ev->get_category()) << "]:";
+			}
+
+			sinsp_threadinfo* p_thr = thread->get_parent_thread();
+			int64_t parent_pid = -1;
+			if(nullptr != p_thr)
+			{
+				parent_pid = p_thr->m_pid;
+			}
+
+			cout << "[PPID=" << parent_pid << "]:"
+			     << "[PID=" << thread->m_pid << "]:"
+			     << "[TYPE=" << get_event_type(ev->get_type()) << "]:"
+			     << "[EXE=" << thread->get_exepath() << "]:"
+			     << "[CMD=" << cmdline << "]"
+			     << endl;
+		}
+	}
+	else
+	{
+		cout << "[EVENT]:[" << get_event_category(ev->get_category()) << "]:"
+		     << ev->get_name() << endl;
+	}
 }
 
 void json_dump(sinsp& inspector)
 {
-    // Initialize JSON formatters
-    static sinsp_evt_formatter* default_formatter = new sinsp_evt_formatter(&inspector, DEFAULT_OUTPUT_STR);
-    static sinsp_evt_formatter* process_formatter = new sinsp_evt_formatter(&inspector, PROCESS_DEFAULTS);
-    static sinsp_evt_formatter* net_formatter = new sinsp_evt_formatter(&inspector, PROCESS_DEFAULTS " %fd.name");
-
-    sinsp_evt* ev = get_event(inspector, [](const std::string& error_msg) {
-        cout << R"({"error": ")" << error_msg << R"("})" << endl;
-    });
-
-    if (ev == nullptr)
-    {
-        return;
-    }
-
-    std::string output;
-    sinsp_threadinfo* thread = ev->get_thread_info();
-
-    if (thread)
-    {
-        if(g_all_threads || thread->is_main_thread())
-        {
-            if(ev->get_category() == EC_PROCESS)
-            {
-                process_formatter->tostring(ev, output);
-            }
-            else if(ev->get_category() == EC_NET || ev->get_category() == EC_IO_READ || ev->get_category() == EC_IO_WRITE)
-            {
-                net_formatter->tostring(ev, output);
-            }
-            else
-            {
-                default_formatter->tostring(ev, output);
-            }
-        }
-        else
-        {
-            // Prevent empty lines from being printed
-            return;
-        }
-    }
-    else
-    {
-        default_formatter->tostring(ev, output);
-    }
-
-    cout << output << std::endl;
+	// Initialize JSON formatters
+	static sinsp_evt_formatter* default_formatter = new sinsp_evt_formatter(&inspector, DEFAULT_OUTPUT_STR);
```

Comment:
Seems initialized twice, keep the one with `static`.

Reply:
ty :)